### PR TITLE
Added some type hinting

### PR DESCRIPTION
**Pull request by BCurbs**
I added a bunch of type hinting, and then tried to fix flake8, but it looks like it was already messed up so I'm submitting as is. 
Should go a long way to fixing #286

### CODE REVIEW

**Comment by theOehrly** [comment]
Thank you for submitting this PR. I haven't reviewed this properly yet. I'll try to get around to it soon.
Regarding flake8, line length compliance is only enforced for changed lines. Thereby, FastF1 should slowly transition to full compliance with that guideline. You don't need to fix that except on lines that you modify anyways.

**Comment by theOehrly** [comment]
@BCurbs in general, I'm happy with these changes. Before I review in detail, please check the docs for how to run the tests and lint checks locally. This will make it much easier to get the CI to pass.

**Review by theOehrly — CHANGES_REQUESTED** [review]
I finally got around to taking a proper look at this. 

Some things that need to be done still:
- lint failure: the flake8 check needs to pass. You can also run this locally. Check the docs to see how
- When type hints are present, no type information should be given in docstrings. Therefore, you should remove the type information from all docstrings where type hints have been added
- I've added some more specific notes as review comments

File: fastf1/core.py [hunk 1]
```diff
@@ -2020,18 +2021,18 @@ def get_telemetry(self):
         pos_data = self.get_pos_data(pad=1, pad_side='both')
         car_data = self.get_car_data(pad=1, pad_side='both')
 
-        # calculate driver ahead from from data without padding to
+        # calculate "driver ahead from" from data without padding to
```
[the remark this answers is not in the document]
Comment (BCurbs) [reply fastf1/core.py:2024]: My IDE got real angry with me for a grammar error before.

**Comment by theOehrly** [comment]
@BCurbs just FYI, I haven't forgotten about this. I'm just really busy right now with other things. Hopefully I'll get around to looking at your PR again somewhere next week.

**Review by theOehrly — CHANGES_REQUESTED** [review]
Hey, finally I got around to this again. Mostly everything looks good. I've added a few comments, nothing major. Especially in core.py there are a few methods where you have added type hints to the arguments or as return type but not removed the information from the docstring. I guess you just missed those, because, for example, in events.py this is correct.

**Comment by theOehrly** [comment]
One more thing, please merge with master, because I needed to fix the flake8 CI test. Else it won't work here.

**Comment by BCurbs** [comment]
The email associated with the commits has switched to a different github account, which is going to be my primary  account after this pull request is complete. Other than that, I think I have everything fixed.

File: fastf1/events.py [hunk 2]
```diff
@@ -153,6 +153,7 @@
 """  # noqa: W605 invalid escape sequence (escaped space)
 import collections
 import datetime
+from datetime import datetime
```
Comment (theOehrly) [inline fastf1/events.py:156]: This causes the test failure. You should not change that import

File: fastf1/events.py [hunk 3]
```diff
@@ -398,15 +399,15 @@ def get_event_schedule(year: int, *, include_testing: bool = True, force_ergast:
 
 
 def get_events_remaining(
-        dt=None, *, include_testing: bool = True,
+        dt: Optional[datetime] = None, *, include_testing: bool = True,
```
Comment (theOehrly) [inline fastf1/events.py]: use Optional[datetime.datetime] here instead

**Comment by theOehrly** [comment]
It'll be easier for you to catch these errors if you set up the tests to run locally. Especially because I need to approve each CI run here as you are a first-time contributor to this repository,
See https://theoehrly.github.io/Fast-F1/contributing/index.html

**Comment by BCurbs** [comment]
Yep, you're good to push your changes.

**Comment by theOehrly** [comment]
Thank you very much for all your work on this PR!

**Comment by transorsmth** [comment]
Thank you for your review, I haven't worked with any of these things like dataframes or anything in this structure yet, so thanks for putting up with me.

**Comment by theOehrly** [comment]
> Thank you for your review, I haven't worked with any of these things like dataframes or anything in this structure yet, so thanks for putting up with me.

No worries, with changes throughout the project, this was not necessarily a simple PR. Also, my aim for this project is to encourage more contributions. And reviewing PRs is an interesting challenge and learning opportunity for myself, as I am fairly new to that.

**Pull request merged**